icu4c/source/common/uloc.cpp

```diff
@@ -1486,7 +1472,11 @@ uloc_openKeywords(const char* localeID,
     }
 
     /* Skip the language */
-    ulocimp_getLanguage(tmpLocaleID, NULL, 0, &tmpLocaleID);
+    ulocimp_getLanguage(tmpLocaleID, &tmpLocaleID, *status);
+    if (U_FAILURE(*status)) {
```

Comment:
optional: It's ok (and a little less clutter) to write this as a one-liner (but still with the braces): `if (U_FAILURE(*status)) { return 0; }`